Add examples and clarified some sections in Beyla component (#2975)
grafana/alloy: docs/sources/reference/components/beyla/beyla.ebpf.md

The component exposes metrics that can be collected by a Prometheus scrape component, and traces that can be forwarded to an OTel exporter component.

{{< admonition type="note" >}}

> To run this component, {{< param "PRODUCT_NAME" >}} requires administrative privileges, or at least it needs to be granted the following capabilities: `BPF`, `SYS_PTRACE`, `NET_RAW` `CAP_CHECKPOINT_RESTORENET_RAW`, `DAC_READ_SEARCH`, and `PERFMON`. The amount of required capabilities depends on the specific use case, refer to [Beyla capabilities][] for more information.

Reviewer: ```suggestion
To run this component, {{< param "PRODUCT_NAME" >}} requires administrative privileges, or at least it needs to be granted the following capabilities: `BPF`, `SYS_PTRACE`, `NET_RAW` `CAP_CHECKPOINT_RESTORENET_RAW`, `DAC_READ_SEARCH`, and `PERFMON`.
The number of required capabilities depends on the specific use case.
Refer to the [Beyla capabilities][] for more information.
```

| `exclude` | `list(string)` | List of attributes to exclude.                         | `[]`    | no       |
| `include` | `list(string)` | List of attributes to include. Use `*` to include all. | `[]`    | no       |

> `include` is a list of attributes that need to be reported. Each attribute can be an attribute name or a wildcard (for example, `k8s.dst.*` to include all the attributes starting with `k8s.dst`).

Reviewer: ```suggestion
`include` is a list of attributes that need to be reported.
Each attribute can be an attribute name or a wildcard, for example, `k8s.dst.*` to include all the attributes starting with `k8s.dst`.
```

> `exclude` is a list to of attribute names/wildcards containing the attributes to remove from the `include` list (or the default attribute set).

Reviewer: ```suggestion
`exclude` is a list to of attribute names/wildcards containing the attributes to remove from the `include` list, or from the default attribute set.
```

The following example shows how you can include and exclude specific attributes:

```alloy
attributes {
  select {
      attr = "sql_client_duration"
      include = ["*"]
      exclude = ["db_statement"]
  }
}
```

> Additionally, you can use `*` wildcards as metric names to add and exclude attributes for groups of metrics having the same name. For example:

Reviewer: ```suggestion
Additionally, you can use `*` wildcards as metric names to add and exclude attributes for groups of metrics having the same name.
For example:
```

```alloy
attributes {
  select {
      attr = "http_*"
      include = ["*"]
      exclude = ["http_path", "http_route"]
  }
  select {
      attr = "http_client_*"
      // override http_* exclusion
      include = ["http_path"]
  }  
}
```

> In the previous example, all the metrics with a name starting with `http_` (or `http`.) would include all the possible attributes but `http_path` and `http_route` (or `http.path`/`http.route`). The `http_client_*` section would override the base configuration, enabling the `http_path` attribute for the HTTP client metrics and `http_route` for the HTTP server metrics.

Reviewer: ```suggestion
In the previous example, all the metrics with a name starting with `http_` or `http.` would include all the possible attributes but `http_path` and `http_route` or `http.path` and `http.route`.
The `http_client_*` section would override the base configuration, enabling the `http_path` attribute for the HTTP client metrics and `http_route` for the HTTP server metrics.
```

| `high_request_volume`         | `bool`        | Optimize for immediate request information when response is seen.          | `false` | no       |
| `heuristic_sql_detect`        | `bool`        | Enable heuristic-based detection of SQL requests.                         | `false` | no       |

> `enable_context_propagation` enables context propagation using Linux Traffic Control probes. To know more about this topic, read [Distributed traces with Beyla][]

Reviewer: ```suggestion
`enable_context_propagation` enables context propagation using Linux Traffic Control probes.
For more information about this topic, refer to [Distributed traces with Beyla][]
```

### `filters`

> The `filters` block allows filtering both application and network metrics by attribute values.

Reviewer: ```suggestion
The `filters` block allows you to filter both application and network metrics by attribute values.
```

> For a list of metrics under the application and network family, as well as their attributes, check the [Beyla exported metrics][].

Reviewer: ```suggestion
For a list of metrics under the application and network family, as well as their attributes, refer to the [Beyla exported metrics][].
```
